### PR TITLE
Adding ability to switch between front and rear facing camera.

### PR DESCRIPTION
**Pull request by kevalpatel2106**
- This will allow the user to switch between front and rear-facing camera and record the video by adding one camera toggle button.  (See the screen shot.)
Camera toggle switch

- The developer can also open front-facing camera by default by passing `VideoCaptureActivity.EXTRA_CAMERA_FACING` parameter. 

- The developer can also disable camera toggle switch by passing false to `allowFrontFacingCamera` while creating `CaptureConfiguration`. By default, camera toggle switch is on.

### CODE REVIEW

**Comment by JeroenMols** [comment]
Awesome @kevalpatel2106 !
I'm a bit busy currently, but I will try to merge things next week.

**Comment by kevalpatel2106** [comment]
@JeroenMols Thank you very much.

File: library/src/main/java/com/jmolsmobile/landscapevideocapture/VideoCaptureActivity.java [hunk 1]
```diff
@@ -130,14 +145,20 @@ public void onDeclineButtonClicked() {
...
+
+    @Override
     public void onToggleCamera() {
-     try {
-         mVideoRecorder.toggleCamera();
-     } catch (Exception e) {
-         e.printStackTrace();
-     }
- }
+        //restart the activity with other camera. That is front facing if the current camera is rear facing and
```
Comment (JeroenMols) [inline library/src/main/java/com/jmolsmobile/landscapevideocapture/VideoCaptureActivity.java]: Is restarting the activity really necessary? Or would it also work without?

File: library/src/main/java/com/jmolsmobile/landscapevideocapture/view/VideoCaptureView.java [hunk 2]
```diff
@@ -84,31 +85,24 @@ private void initialize(Context context) {
...
-        }
-
-        return false;
-    }
-
     public void setRecordingButtonInterface(RecordingButtonInterface mBtnInterface) {
         this.mRecordingInterface = mBtnInterface;
     }
 
+    public void setCameraFacing(boolean isFrontFacing){
```
Comment (JeroenMols) [inline library/src/main/java/com/jmolsmobile/landscapevideocapture/view/VideoCaptureView.java]: If we always restart the activity, then a setter shouldn't be necessary, right?

File: README.md [hunk 3]
```diff
@@ -78,6 +79,7 @@ This library provides a full and reusable custom camera, which:
 ```java
          final Intent intent = new Intent(getActivity(), VideoCaptureActivity.class);
          intent.putExtra(VideoCaptureActivity.EXTRA_CAPTURE_CONFIGURATION, config);
+         intent.putExtra(VideoCaptureActivity.EXTRA_CAMERA_FACING, PredefinedCameraFacing.REAR_FACING); //Opens rearfacing camera by default.
```
Comment (JeroenMols) [inline README.md]: Would not expose this via an intent, but only keep it in the Capture configuration. -> Good apis only have one way to achieve a certain function and the camera option seems to fit best with the capture configuration.

**Comment by JeroenMols** [comment]
Awesome @kevalpatel2106 and @danielgomezrico !

Overall:
- Very much appreciate the short concise commits
- Thanks for sticking to the same coding style
- Very cool animation! Is that why you restart the activity of was there another limitation?
- Commits look pretty good, and all test are running -> Am going to do a final check to see if the bigger picture still "fits"
- Added few comments to the commit
- There is an issue where my preview is  upside down (nexus 6p). -> I'll look into that

Will try to wrap this up asap, sorry for the delay already gentlemen!

**Comment by JeroenMols** [comment]
Hi guys, I did some minor refactors and merged the code into master.
Thanks again for your awesome contribution @kevalpatel2106  and @danielgomezrico. 

Could you install the latest app from Google play and test if the orientation is fine in portrait?

**Comment by kevalpatel2106** [comment]
> Could you install the latest app from Google play and test if the orientation is fine in portrait?

@JeroenMols Sure. Let me try the app from the google play.

> Very cool animation! Is that why you restart the activity of was there another limitation?

Actually, I was not able to switch between front and rear-facing camera while the preview is running continually. So, I looked into the Google Camera app and I figure out that they are restarting the activity while changing the camra. So, I did the same thing and code works. I added the animations to override default android activity transition animation as they were not looking pretty cool. :-)

Thank you for your feedbacks.

**Comment by JeroenMols** [comment]
@kevalpatel2106 I just fixed an issue this morning -> after camera switch, a new activity was started on top of the old one. When confirming recording the result however would not be propagated back and you would get a "recording failed".

Fixed that this morning, so you'll have to wait for the 1.2.2 app to be live on Google play.

**Comment by kevalpatel2106** [comment]
@JeroenMols Ok. I'll wait for that.

**Comment by JeroenMols** [comment]
@kevalpatel2106 App should be live. (had to uninstall it to get the updated version though)

**Comment by kevalpatel2106** [comment]
@JeroenMols I tested it. It is working as expected in my device Motorola G3 Android 6.0.1.

**Comment by JeroenMols** [comment]
Awesome!

**Comment by caipivara** [comment]
@kevalpatel2106 @kevalpatel2106 Guys I tested on Nexus 5X and the camera is upside down too.

**Comment by kevalpatel2106** [comment]
@caipivara Now it's resolved, right?

**Pull request merged**